paseos/communication/radio_transmitter_model.py

```diff
@@ -0,0 +1,70 @@
+import math
+
+from loguru import logger
+
+from .transmitter_model import TransmitterModel
+from ..utils.gain_calc import calc_radio_gain_from_wavelength_diameter
+
+
+class RadioTransmitterModel(TransmitterModel):
```

Comment:
As a general (and maybe educational aspect), in terms of software design, I thin,k you are overusing classes and inheritance. The majority of the code in each subclass is a duplication of code in other files, the functionality of the different classes is quite similar and the created classes are not really necessary since you are not actually introducing any state (meaning the classes don't change over the course of the simulation / behave differently).

You could vastly simplify the code by storing the (constant) parameters of the comm link / transmitter / receiver in a dictionary inside the actor, then just pass the actor to the function that computes bitrate?

Reply:
I agree that I made it slightly to complicated with the receiver, transmitter and link subclasses. Simplified that to simply ReceiverModel, TransmitterModel and LinkModel. I guess these could be a dictionary inside the actor, though the models do keep track of their active state and the bitrate histories. Additionally, I find this a lot cleaner than a large dictionary or DotMap.